Packs/iDefense/Integrations/iDefense_v2/iDefense_v2.py

```diff
@@ -310,14 +318,190 @@ def uuid_command(client: Client, args: dict, reliability: DBotScoreReliability)
             'TypeOfUse': res.get('last_seen_as', ''),
             'LastPublished': str(last_published_format)
         }
+        analysis_info = _enrich_analysis_result_with_intelligence(analysis_info, doc_search_client)
+
     return CommandResults(indicator=indicator,
                           raw_response=res,
                           readable_output=tableToMarkdown('Results', analysis_info))
 
 
+def hash_command(client: Client, args: dict, reliability: DBotScoreReliability, doc_search_client: Client) -> CommandResults:
+    """
+    Takes the hash passed by user, determines the type of hash and does an API call.
+
+    Args:
+        client: iDefense client.
+        args: arguments obtained with the command representing the indicator value to search.
+        reliability: reliability of the source.
+
+    Returns:
+        CommandResults containing the indicator, the response and a readable output.
+    """
+    try:
+        hash: str = str(args.get('file'))
+        key_type: str = ""
+
+        # For MD5   --->
+        if len(hash) == 32:
+            key_type = 'key'
+            printKey = 'MD5'
+        # For SHA1  --->
+        elif len(hash) == 40:
+            key_type = 'sha1'
+            printKey = 'SHA1'
+        # For SHA256   --->
+        elif len(hash) == 64:
+            key_type = 'sha256'
+            printKey = 'SHA256'
+
+        if key_type == "":
+            return CommandResults(indicator=None, raw_response={},
+                                  readable_output=f"Invalid hash value: {hash}")
+        else:
+            Res = client.threat_indicator_search(url_suffix=f'/v0?files.{key_type}.query={hash}&page_size=1')
+            Res, analysis_info, context = _hash_extract(Res, reliability, key_type, hash, doc_search_client)
+            return CommandResults(raw_response=Res,
+                                  outputs=context,
+                                  readable_output=tableToMarkdown(f'ACTI results for {printKey} hash: {hash}', analysis_info))
+
+    except Exception as e:
+        if 'Failed to parse json object from response' in e.args[0]:
+            return CommandResults(indicator=None, raw_response={},
+                                  readable_output=f"No results were found for hash {hash}")
+        else:
+            raise e
+
```

Comment:
Can you tell an invalid json reseponse from an empty one?

Reply:
Have changed to this ➡️ 
![image](https://user-images.githubusercontent.com/40510780/146362821-47cb07d7-c643-4594-892e-ba54197caa5c.png)

Reply:
if you give some improper hash  it will show this to the user :
![image](https://user-images.githubusercontent.com/40510780/146363002-8bbda64f-d046-46ab-8ab5-ecfe366c6b01.png)

Reply:
I see, we recommend that you return the reason too (whether the hash is invalid or there are no results) so users know what's wrong and are able to debug.

Reply:
It's also recommended to catch specific exceptions rather than any exception, but not mandatory.

Reply:
I think users will get to know what error they are facing, if it's not an MD5/SHA1/SHA256 it will display
`Invalid hash value: `
and when they provide a hash of proper format but if we don't have intelligence related to that hash , it will display
`No results were found for hash `